### PR TITLE
Ensure the dialog does not close if you drag outside by mistake

### PR DESCRIPTION
**Pull request by echarles**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/11672

## Code changes

Add internal state in Dialog class to identify where the last mousedown occured.

## User-facing changes

Before

before

After

after

## Backwards-incompatible changes

None

### CODE REVIEW

File: packages/apputils/src/dialog.tsx [hunk 1]
```diff
@@ -206,10 +207,14 @@ export class Dialog<T> extends Widget {
    * not be called directly by user code.
    */
   handleEvent(event: Event): void {
+    console.log('---', event);
```
Comment (jtpio) [inline packages/apputils/src/dialog.tsx:210]: Looks like we can remove this `console.log`:

```suggestion
```

**Review by fcollonval — APPROVED** [review]
Thanks @echarles

**Comment by jtpio** [comment]
@meeseeksdev please backport to 3.2.x

**Comment by echarles** [comment]
Thx @jtpio and @fcollonval for the reviews and merge 👍

**Pull request merged**